### ISSUE TITLE
303->310 should be 309 instead of 310

### ISSUE THREAD

**Issue by adamcsills** [maltepet/lightcalc-destiny#6]
Using:
http://mattaltepeter.com/destiny/

Original item is a 303 legendary
Infusion fuel is a 310 exotic

The website says the result is a 310. I tested the inf() function outside the site and it correctly returns a 309. I haven't gotten a chance to dig in to see why it's wrong, just reporting it.

**Comment by maltepet** [comment]
Yeah, the thresholds are just wrong. I think once you get above 300, the game adjusts the max difference at which they will give you 100% of the difference.

**Comment by adamcsills** [comment]
Using the calculation in inf() it's right, but the website is showing 310.

303 -> 310 is 5.6, rounded up ends up with a 309. That uses your inf() calculation exactly.

**Comment by adamcsills** [comment]
So it's not a threshold issue; the code you have here is different than the code on your website.

**Comment by adamcsills** [comment]
Found it. The code on the website has this:

var threshold = getThreshold(rarity);
// ...
if (diff <= threshold) {

This is wrong. It should be diff < threshold. It gives 6 min on legendary, not 7.

**Comment by maltepet** [comment]
Oh, yeah. I've been bad about pushing stuff to github.

**Comment by maltepet** [comment]
great catch. I've fixed it on the website

**Comment by adamcsills** [comment]
Good deal, thanks. A buddy was sad he didn't get his gun to 310 like he
thought he would, glad to prevent future sadness :)
